core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -341,12 +340,13 @@ public void onTimeout(TimeValue timeout) {
                     }
                     // use null timeout as we use timeout on the latchwait
                 }, MasterNodeChangePredicate.INSTANCE, null);
-            }
-
-            try {
-                latch.await(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.get(settings).millis(), TimeUnit.MILLISECONDS);
-            } catch (InterruptedException e) {
-                throw new ElasticsearchTimeoutException("Interrupted while waiting for initial discovery state");
+                try {
+                    if (latch.await(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.get(settings).millis(), TimeUnit.MILLISECONDS)) {
```

Comment:
This should be `== false` right? otherwise we will warn if the latch *found* a state